src/python/dxpy/dx_extract_utils/InputsValidator.py

```diff
@@ -171,103 +179,95 @@ def interpret_mutually_exclusive_group(self, check):
-#                 expected_exceptions=(
-#                     InvalidInput,
-#                     InvalidState,
-#                 ),
-#             )
-#         except Exception as details:
-#             err_exit(str(details))
-
-#     def assure_dataset_version(self):
-#         # checking cohort/dataset version
-#         if self.http_request["datasetVersion"] != "3.0":
-#             err_exit(
-#                 "%s : Invalid version of cohort or dataset. Version must be 3.0"
-#                 % self.http_request["datasetVersion"]
-#             )
-
-#     def resolve_dataset_project(self):
-#         # Defining dataset project
-#         if ("Dataset" in self.http_request["recordTypes"]) or (
-#             "CohortBrowser" in self.http_request["recordTypes"]
-#         ):
-#             self.dataset_project = self.http_request["datasetRecordProject"]
-#         else:
-#             err_exit(
-#                 "%s : Invalid path. The path must point to a record type of cohort or dataset"
-#                 % self.http_request["recordTypes"]
-#             )
-
-#     def get_http_request_info(self):
-#         return self.http_request
+class PathValidator:
+    """
+    PathValidator class checks for invalid object inputs and its combination with passed arguments.
+    """
+
+    def __init__(self, parser_dict, project, entity_result, error_handler=print):
+        self.parser_dict = parser_dict
+        self.project = project
+        self.entity_result = entity_result
+        self.error_handler = error_handler
+
+        self.record_http_request_info = None
+
+    def throw_error(self, message):
+        self.error_handler(message)
+
+    def try_populate_record_http_request_info(self):
+        # record_http_request_info contains crucial information for records
+        try:
+            self.record_http_request_info = DXHTTPRequest(
+                "/" + self.entity_result["id"] + "/visualize",
+                {"project": self.project, "cohortBrowser": False},
+            )
+        except (InvalidInput, InvalidState):
+            self.throw_error(
+                "Invalid cohort or dataset: {}".format(self.entity_result["id"]),
+            )
+        except Exception as details:
+            self.throw_error(str(details))
+
+    def is_object_in_current_project(self):
+        # for object in a different project:
+        if self.project != self.entity_result["describe"]["project"]:
+            self.throw_error(
+                'Unable to resolve "{}" to a data object or folder name in {}. Please make sure your object is in your selected project.'.format(
+                    self.parser_dict.get("path"), self.project
+                )
+            )
+
+    def is_cohort_or_dataset(self):
+        # resolving non record/cohort type
+        if self.entity_result is None:
+            self.throw_error(
+                "The path must point to a record type of cohort or dataset, not a folder."
```

Comment:
Does the empty `self.entity_result` always mean the path is a folder? (that is what the message implies)